File: src/ui/app/root/application/models/Attendance/AttendenceMaster.ts
import { StringifyOptions } from "querystring";
import { Time } from "highcharts";

export interface IAttendanceAttendenceMaster {
    attendenceMasterId: string;
    timeTableId: string;
    dated: Date;
    isApproved: boolean;
    statusId: number;
    loggerId: string;
    operation: string;
}

export interface IOperationAttendanceMaster {
    insertedBy: number;
    browserInfo: string;
    insertionTime: string;
    inTime: boolean;
    approvalTime: string;
    approvedBy: number;
}

export interface IAttendanceAttendenceMasterVM {
    attendenceMasterId: string;
    timeTableId: string;
    dated: string;
    isApproved: boolean;
    statusId: number;
    loggerId: string;
    sectionName: string;
    name: string;
    roomName: string;
    dayName: string;
    staffName: string;
    startTime: string;
    endTime: string;
    session: string;
    campusId: string;
    campusName: string;
    sessionId: string;
    sectionId: string;
    programDetailId: string;
    // campusProgramId: string;
    classId: string;
}

export interface IAttendanceApprovalDataVM {
    attendenceMasterId: string;
    timeTableId: string;
    dated: string;
    date: string;
    isApproved: boolean;
    statusId: number;
    loggerId: string;
    fullName: string;
    campusId: string;
    sessionId: string;
    programDetailId: string;
    courseId: string;
    sectionName: string;
    dayName: string;
    operation: string;
}

export interface IAttendanceApprovalDataExVM {
    attendenceMasterId: string;
    timeTableId: string;
    dated: string;
    date: string;
    isApproved: boolean;
    statusId: number;
    loggerId: string;
    course: string;
    campusId: string;
    sessionId: string;
    programDetailId: string;
    courseId: string;
    sectionName: string;
    dayName: string;
    operation: string;
    staffName: string;
    startTime: string;
    endTime: string;
}
export interface AttendanceApprovalExNotificationVM {
    attendenceMasterId: string;
    dated: string;
    course: string;
    sessionId: string;
    campusId: string;
    campusProgramId: string;
    classId: string;
    sectionCourseLinkId: string;
    rollNo: string;
    code: string;
    name: string;
}


export interface AttendenceMonthWise {
    iD: string;
    month: string;
    total: number;
    presents: number;
    leave: number;
    startDate: Date;
    endDate: Date;
    absents: number;
    attendencePercentage: number;
    aggrePercentage: number;
}

export interface AttendencePercentages {
    iD: string;
    total: number;
    presents: number;
    leave: number;
    absents: number;
    attendencePercentage: number;
  
}


export interface VWAttendanceHistory {
    attendenceMasterId: string;
    dated: string;
    inTime: string;
    approvedBy: string;
    insertedBy: string;
    browserInfo: string;
    approvalTime: string;
    insertionTime: string;
    statusId: number;
    startTime: string;
    endTime: string;

}
export interface AttendanceDevice {

    browserInfo: string;
    count: number;
    dated: string;
}
export interface IDayCloseInsert {
    attendenceMasterId: string;
    operation: string;
}

export interface IOperationDayClose {
    insertedBy: number;
    browserInfo: string;
    insertionTime: string;
}

export interface StudentToSend {
    id: string;
    phoneNo: string;
    fullName: string;
    programCourseLinkId: string;
    sectionCourseLinkId: string;
    courseName: string;
    admissionFormId: string;
}







export interface IAttendanceAttendanceStudentInfo {
    newID: string;
    admissionFormId: string;
    studentName: string;
    dated: string;
    dayName: string;
    refferenceNo: string;
    campusName: string;
    sessionName: string;
    rollNo: string;
    description: string;
    programName: string;
    sectionName: string;
    className: string;
    shiftName: string;
    courseName: string;
    attendanceStatus: string;
}

export interface DeviceInfoEx {
    operation: string;
}


export interface SmartAttendence {
    id: string;
    cityId: string;
    cityName: string;
    subCityId: string;
    subCityName: string;
    campusId: string;
    campusName: String;
}

File: src/data/Model/Cms360/Attendance/AttendenceMaster.cs
/*
*   Auther: H.Muhammad Kamran
*   email: hmuhdkamran@gmail.com
*   contact: +92 (313 / 333) 9112 845
*/

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Cms360.Data.Model
{
    [Table("AttendenceMaster", Schema = "Attendance")]
    public partial class AttendanceAttendenceMaster
    {
        [Key]
        [Required]
        public Guid AttendenceMasterId { get; set; }
        [Required]
        public Guid TimeTableId { get; set; }
        [Required]
        public DateTime Dated { get; set; }
        [Required]
        public bool IsApproved { get; set; }

        [Column(TypeName = "jsonb")]
        public string Operation { get; set; }

        [Required]
        public int StatusId { get; set; }
        [Required]
        public Guid LoggerId { get; set; }

    }

    [Table("VWAttendenceMasterVM", Schema = "Attendance")]
    public partial class AttendanceAttendenceMasterVM
    {
        [Key]
        [Required]
        public Guid AttendenceMasterId { get; set; }
        [Required]
        public Guid TimeTableId { get; set; }
        [Required]
        public DateTime Dated { get; set; }
        [Required]
        public bool IsApproved { get; set; }
        [Required]
        public int StatusId { get; set; }
        [Required]
        public Guid LoggerId { get; set; }

        // public string FullName { get; set; }
        public string SectionName { get; set; }

        public string Name { get; set; }

        public string RoomName { get; set; }

        public string DayName { get; set; }

        public string StaffName { get; set; }

        public TimeSpan StartTime { get; set; }

        public TimeSpan EndTime { get; set; }
        public string Session { get; set; }

        public Guid CampusId { get; set; }

        public string CampusName { get; set; }

        public Guid SessionId { get; set; }

        public Guid SectionId { get; set; }

        public Guid ProgramDetailId { get; set; }

        // public Guid CampusProgramId { get; set; }

        public Guid ClassId { get; set; }

    }

    [Table("VWAttendanceApprovalData", Schema = "Attendance")]
    public partial class AttendanceApprovalDataVM
    {
        [Key]

        public Guid AttendenceMasterId { get; set; }
        public Guid TimeTableId { get; set; }
        public DateTime Dated { get; set; }
        public string Date
        {
            get
            {
                if (this.Dated != null)
                {
                    return this.Dated.ToShortDateString();
                }
                else
                {
                    return null;
                }
            }
        }
        public bool IsApproved { get; set; }
        public int StatusId { get; set; }
        public Guid LoggerId { get; set; }
        public string FullName { get; set; }
        public Guid CampusId { get; set; }
        public Guid SessionId { get; set; }
        public Guid ProgramDetailId { get; set; }
        public Guid CourseId { get; set; }
        public string SectionName { get; set; }
        public string DayName { get; set; }
        [Column(TypeName = "jsonb")]
        public string Operation { get; set; }

    }

    [Table("VWAttendanceApprovalDataEx", Schema = "Attendance")]
    public partial class AttendanceApprovalDataExVM
    {
        [Key]

        public Guid AttendenceMasterId { get; set; }
        public Guid TimeTableId { get; set; }
        public DateTime Dated { get; set; }
        public string Date
        {
            get
            {
                if (this.Dated != null)
                {
                    return this.Dated.ToShortDateString();
                }
                else
                {
                    return null;
                }
            }
        }
        public bool IsApproved { get; set; }
        public int StatusId { get; set; }
        public Guid LoggerId { get; set; }
        public string Course { get; set; }
        public Guid CampusId { get; set; }
        public Guid SessionId { get; set; }
        public Guid ProgramDetailId { get; set; }
        public Guid CourseId { get; set; }
        public string SectionName { get; set; }
        public string DayName { get; set; }
        [Column(TypeName = "jsonb")]
        public string Operation { get; set; }
        public string StaffName { get; set; }
        public TimeSpan StartTime { get; set; }
        public TimeSpan EndTime { get; set; }
    }
        public class AttendenceMonthWise
    {  [Key]

        public Guid ID { get; set; }
        public string Month { get; set; }

        public int Presents { get; set; }
        public int Total { get; set; }

        public int Absents { get; set; }
        public int Leave { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public float AttendencePercentage { get; set; }

        public float AggrePercentage { get; set; }

    }


    public class AttendanceApprovalExNotificationVM
    {
         [Key]
         public Guid AttendenceMasterId { get; set; }
         public DateTime Dated { get; set; }
         public string Course { get; set; }
         public Guid sessionId { get; set; }
         public Guid CampusId { get; set; }
         public Guid CampusProgramId { get; set; }
         public Guid ClassId { get; set; }
         public Guid SectionCourseLinkId { get; set; }
         public string RollNo { get; set; }
         public string Code { get; set; }
         public string Name { get; set; }

    }


    [Table("VWAttendanceHistory", Schema = "Attendance")]
    public class VWAttendanceHistory
    {
        [Key]
        public Guid AttendenceMasterId { get; set; }
        public DateTime Dated { get; set; }
        public string InTime { get; set; }
        public string ApprovedBy { get; set; }
        public string InsertedBy { get; set; }
        public string BrowserInfo { get; set; }
        public string ApprovalTime { get; set; }
        public string InsertionTime { get; set; }
        public int StatusId { get; set; }
        public TimeSpan StartTime { get; set; }
        public TimeSpan EndTime { get; set; }

    }

    public class AttendanceDevice
    {
        [Key]
        public string BrowserInfo { get; set; }
        public int Count { get; set; }
        public DateTime Dated { get; set; }

    }

    public class StudentToSend
    {
        [Key]
        public Guid Id { get; set; }
        public string PhoneNo { get; set; }
        public string FullName { get; set; }
        public Guid ProgramCourseLinkId { get; set; }
        public Guid SectionCourseLinkId { get; set; }
        public string CourseName { get; set; }
        public Guid AdmissionFormId { get; set; }


    }

    public class DeviceInfoEx

    {
        [Key]

        [Column(TypeName = "jsonb")]
        public string Operation { get; set; }
    }

      public class AttendencePercentages
    {  [Key]

        public Guid ID { get; set; }
        

        public int Presents { get; set; }
        public int Total { get; set; }
        public int Absents { get; set; }
        public int Leave { get; set; }

        public float AttendencePercentage { get; set; }

    

    }


}